RacingMod/Chat/RacingCommands.cs

```diff
@@ -0,0 +1,201 @@
+using System.Text;
+using VRage.Game.ModAPI;
+
+namespace avaness.RacingMod.Chat
+{
+    public class RacingCommands
+    {
+        public const string prefix = "/race";
+
+        private readonly Track race;
+        private readonly Dictionary<string, ChatCommand> commands = new Dictionary<string, ChatCommand>();
+        private string helpMsg, adminHelpMsg;
+
+        public RacingCommands(Track race)
+        {
+            this.race = race;
+
+            MyAPIGateway.Utilities.MessageEntered += MessageEntered;
+            if (RacingConstants.IsServer)
+            {
+                Add(new CmdAutoJoin(), "autojoin", "aj");
+                Add(new CmdJoin(), "join", "j");
+                Add(new CmdLeave(), "leave", "l");
+                Add(new CmdRejoin(), "rejoin");
+                Add(new ACmdClear(), new ACmdFinish(), new ACmdGrant(), new ACmdKick(), new ACmdLaps(), new ACmdLooped(), new ACmdMode(), new ACmdStrictStart(), new ACmdTimers());
+                BuildHelpText();
+                Add(new CmdDebug(), "debug"); // Does not appear in help
+
+                MyAPIGateway.Utilities.MessageRecieved += ReceiveCommand;
+                RacingSession.Instance.Net.Register(RacingConstants.packetCmd, ReceiveCommandPacket);
+            }
+        }
+
+        public void Unload()
+        {
+            MyAPIGateway.Utilities.MessageEntered -= MessageEntered;
+            MyAPIGateway.Utilities.MessageRecieved -= ReceiveCommand;
+        }
+
+        // Called on server
+        private void ReceiveCommandPacket(byte[] data)
+        {
+            try
+            {
+                CommandInfo cmd = MyAPIGateway.Utilities.SerializeFromBinary<CommandInfo>(data);
+                ReceiveCommand(cmd.id, cmd.command);
+            }
+            catch (Exception e)
+            {
+                RacingTools.ShowError(e, GetType());
+            }
+        }
+
+        // Called on server
+        private void ReceiveCommand(ulong id, string command)
+        {
+            IMyPlayer p = RacingTools.GetPlayer(id);
+            if (p != null)
+                ProcessCommand(p, command);
+        }
+
+        // Called on clients
+        private void MessageEntered(string messageText, ref bool sendToOthers)
+        {
+            IMyPlayer p = MyAPIGateway.Session.Player;
+            if (p == null)
+                return;
+            
+            if (IsRaceCommand(messageText))
+            {
+                sendToOthers = false;
+
+                if (RacingConstants.IsServer)
+                    ProcessCommand(p, messageText);
+                else
+                    RacingSession.Instance.Net.SendToServer(RacingConstants.packetCmd, new CommandInfo(messageText, p.SteamUserId));
+            }
+        }
+
+        private bool IsRaceCommand(string command)
+        {
+            return command.StartsWith("/rcd", StringComparison.OrdinalIgnoreCase) || command.StartsWith(prefix, StringComparison.OrdinalIgnoreCase);
```

Comment:
What if I create another mod and register a command `/rcdsomethingelse` ?

Reply:
I was trying to avoid the Split function. Additional checks are needed to make sure there is white space or end of string after.